website/content/en/preview/getting-started/migrating-from-cas/_index.md

```diff
@@ -0,0 +1,326 @@
+
+This guide will also assume you have the `aws` CLI installed.
+You can also perform many of these steps in the console, but we will use the command line for simplicity.
+
+## Create IAM roles
+
+To get started with our migration we first need to create two new IAM roles for nodes provisioned with Karpenter and the Karpenter controller.
+
+To create the Karpenter node role we will use the following policy and commands.
+
+```bash
+echo '{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}' > trust-policy.json
+
+aws iam create-role --role-name KarpenterInstanceNodeRole \
+    --assume-role-policy-document file://trust-policy.json
+```
+Now attach the required policies to the role
+
+```bash
+aws iam attach-role-policy --role-name KarpenterInstanceNodeRole \
+    --policy-arn arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+
+aws iam attach-role-policy --role-name KarpenterInstanceNodeRole \
+    --policy-arn arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
+
+aws iam attach-role-policy --role-name KarpenterInstanceNodeRole \
+    --policy-arn arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+
+# optional for session manager
+aws iam attach-role-policy --role-name KarpenterInstanceNodeRole \
+    --policy-arn arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+```
+
+Now we need to create an IAM role that the Karpenter controller will use to provision new instances.
+The controller will be using [IAM Roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) which requires an OIDC endpoint.
+
+If you have another option for using IAM credentials with workloads (e.g. [kube2iam](https://github.com/jtblin/kube2iam)) your steps will be different.
+
+First we need to get the OIDC endpoint for the cluster.
+```bash
+CLUSTER_NAME=<your existing cluster>
+OIDC_ENDPOINT=$(aws eks describe-cluster --name ${CLUSTER_NAME} \
+    --query "cluster.identity.oidc.issuer" --output text)
+ACCOUNT_NUMBER=$(aws sts get-caller-identity --query 'Account' \
+    --output text)
+```
+
+Use that information to create our IAM role, inline policy, and trust relationship.
+```bash
+echo "{
+    \"Version\": \"2012-10-17\",
+    \"Statement\": [
+        {
+            \"Effect\": \"Allow\",
+            \"Principal\": {
+                \"Federated\": \"arn:aws:iam::${ACCOUNT}:oidc-provider/${OIDC_ENDPOINT#*//}\"
+            },
+            \"Action\": \"sts:AssumeRoleWithWebIdentity\",
+            \"Condition\": {
+                \"StringEquals\": {
+                    \"${OIDC_ENDPOINT#*//}:aud\": \"sts.amazonaws.com\",
+                    \"${OIDC_ENDPOINT#*//}:sub\": \"system:serviceaccount:karpenter:karpenter\"
+                }
+            }
+        }
+    ]
+}" > trust-policy.json
```

Comment:
can this file be called something different since it will override the previous `trust-policy.json`?